api/errors/builder.go

```diff
@@ -0,0 +1,93 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker/builder component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorCodeAtLeastOneArg is generated when the parser comes across a
+	// Dockerfile command that doesn't have any args.
+	ErrorCodeAtLeastOneArg = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
I mean that it makes more sense to define a few classes of errors and divide them with solid messaging, than to make a code for every condition. Effectively, it is easier to add them than to taken them away.

In this case, we really want to have an error code not for each condition, but rather, for each action a consumer of these functions can take action upon. We need to ask the question, can a client or callee take different action based receiving a `EXACTLYONEARG` or `ATLEASTONEARG`? If not, having different codes for these conditions only serves to make the error handling more complex.